### DISCUSSION TITLE
new source data of type rest doesn't work ??

### DISCUSSION THREAD

**Discussion by musmontis** [Budibase/budibase#2386] [Help]
Hello, 

I want testing budibase to replace retool but I first issue. I cannot connect my service rest; I have an error
" invalid json response body at {{MY-REST-SERVICE}} reason: Unexpected end of JSON input"
the important point, my service has a basic auth and I configure like that 
![image](https://user-images.githubusercontent.com/3398068/129731361-83c451b0-61ca-4d65-a840-84c5ab69c8b9.png)

I also test another service without basic Auth and everything works.
Do you have an idea of my mistake?

Thanks for your response

**Comment by mjashanks** [comment]
Hey @musmontis - everything looks ok to me. Could you try running exactly the same request in another client - e.g. Postman - to see what the response from the API is?

Also - is your API definitely returning JSON? Budibase expects JSON.

**Reply by musmontis** [reply]
Hey @mjashanks, I tested with my postman and everything is ok, I think the config is bad for my Basic Auth. How can I config correctly my Basic Auth ? In Postman I just field the Authorization tab and choose the authorization type (basic auth) and field user/password.
With BudiBase I field header like this : `Authorization : basic user/passwordbase64`

**Reply by mjashanks** [reply]
ok - so now, using postman:

1. Click "Headers"
2. Show hidden headers
3. You should now be able to see the exact Authorization header that postman is sending - use this for budibase

![image](https://user-images.githubusercontent.com/3524181/130038873-2c5e6a71-21cb-496a-8d86-07f380dec55d.png)

**Comment by musmontis** [comment]
I get the same result 
![image](https://user-images.githubusercontent.com/3398068/130044722-e349180b-6e77-454d-b1f0-5983c2a0ea48.png)
I test budibase in docker environment, I changed anything in the configuration of the docker-compose, maybe the issue comes from there ? 
And I didn't answer the question if my API returns JSON and yes of course
